src/ccbr_tools/jobby.py

```diff
@@ -1,804 +1,272 @@
+    try:
+        if not t or t.strip() == "":
+            return 0
+
+        if "-" in t:
+            days, rest = t.split("-")
+            days = int(days)
+            t = rest
         else:
-            tmp_list.insert(i, insertion_dict[i])
-    return tmp_list
-
-
-def convert_size(size_bytes):
-    """
-    Converts bytes to a human readable format.
-
-    Args:
-        size_bytes (int): Size in bytes to convert.
-
-    Returns:
-        str: Human readable size in the format 'X.YZUNIT'.
-
-    Example:
-        >>> convert_size(1024)
-        '1.0KiB'
-    """
-    # Sizes range from B to YiB,
-    # warning larger sizes storage
-    # may results in blackhole
-    size_name = ("B", "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB")
-    if size_bytes == 0:
-        return "0B"
-    i = int(math.floor(math.log(size_bytes, 1024)))
-    p = math.pow(1024, i)
-    s = round(size_bytes / p, 2)
-    return "{0}{1}".format(s, size_name[i])
-
-
-def to_bytes(size):
-    """
-    Convert a human readable size unit into bytes.
-    Returns None if cannot convert/parse provided size.
-
-    Args:
-        size (str): Human readable size unit to convert.
-
-    Returns:
-        int: Size in bytes.
-
-    Example:
-        >>> to_bytes('1.0KiB')
-        1024
-    """
-    size2bytes = {
-        "b": 1,
-        "bytes": 1,
-        "byte": 1,
-        "k": 1024,
-        "kib": 1024,
-        "kb": 1000,
-        "m": 1024**2,
-        "mib": 1024**2,
-        "mb": 1000**2,
-        "g": 1024**3,
-        "gib": 1024**3,
-        "gb": 1000**3,
-        "t": 1024**4,
-        "tib": 1024**4,
-        "tb": 1000**4,
-        "p": 1024**5,
-        "pib": 1024**5,
-        "pb": 1000**5,
-        "e": 1024**6,
-        "eib": 1024**6,
-        "eb": 1000**6,
-        "z": 1024**7,
-        "zib": 1024**7,
-        "zb": 1000**7,
-        "y": 1024**8,
-        "yib": 1024**8,
-        "yb": 1000**8,
-    }
-
-    size = size.replace(" ", "")
-    match = re.search("(?P<size>[0-9.]+)(?P<units>[a-zA-Z]+)$", size)
-
-    if match:
-        human_units = match.group("units").lower()
-        human_units = human_units.lstrip().rstrip()
-        scaling_factor = size2bytes[human_units]
-        bytes = int(math.ceil(scaling_factor * float(match.group("size"))))
-    else:
-        # Cannot parse units,
-        # cannot convert value
-        # into bytes
+            days = 0
+
+        parts = t.split(":")
+        parts = [float(p) for p in parts]
+
+        if len(parts) == 3:
+            h, m, s = parts
+        elif len(parts) == 2:
+            h = 0
+            m, s = parts
+        elif len(parts) == 1:
+            h = 0
+            m = 0
+            s = parts[0]
+        else:
+            return 0
+
+        total_seconds = int(days) * 86400 + int(h) * 3600 + int(m) * 60 + s
+        return int(round(total_seconds))
+    except Exception:
+        return 0
```

Comment:
probably best to differentiate between invalid numbers and jobs that actually take 0 seconds
```suggestion
        return np.nan
```